src/screencast/wlr_screencast.c

```diff
@@ -97,19 +108,36 @@ static struct wl_buffer *create_shm_buffer(struct xdpw_screencast_instance *cast
 	return buffer;
 }
 
-static void wlr_frame_buffer(void *data, struct zwlr_screencopy_frame_v1 *frame,
+static void wlr_frame_buffer_chparam(struct xdpw_screencast_instance *cast,
 		uint32_t format, uint32_t width, uint32_t height, uint32_t stride) {
-	struct xdpw_screencast_instance *cast = data;
 
-	logprint(TRACE, "wlroots: buffer event handler");
-	cast->wlr_frame = frame;
+	logprint(DEBUG, "wlroots: reset buffer");
 	cast->simple_frame.width = width;
 	cast->simple_frame.height = height;
 	cast->simple_frame.stride = stride;
 	cast->simple_frame.size = stride * height;
 	cast->simple_frame.format = format;
-	cast->simple_frame.buffer = create_shm_buffer(cast, format, width, height,
-		stride, &cast->simple_frame.data);
+	wlr_frame_buffer_clear(cast);
+}
+
+static void wlr_frame_buffer(void *data, struct zwlr_screencopy_frame_v1 *frame,
+		uint32_t format, uint32_t width, uint32_t height, uint32_t stride) {
+	struct xdpw_screencast_instance *cast = data;
+
+	logprint(TRACE, "wlroots: buffer event handler");
+	cast->wlr_frame = frame;
+	if (cast->simple_frame.width != width || cast->simple_frame.height != height || cast->simple_frame.stride != stride || cast->simple_frame.format != format) {
+		logprint(TRACE, "wlroots: buffer properties changed");
+		wlr_frame_buffer_chparam(cast, format, width, height, stride);
+	}
+
+	if (cast->simple_frame.buffer == NULL) {
+		logprint(DEBUG, "wlroots: create shm buffer");
+		cast->simple_frame.buffer = create_shm_buffer(cast, format, width, height,
+			stride, &cast->simple_frame.data);
+	} else {
+		logprint(DEBUG,"wlroots: shm buffer exists");
```

Comment:
Can you make this log a TRACE level log? This will print every single frame where the params haven't changed. I'm trying to keep "per frame" logging events in TRACE.

Reply:
Shure. This log was only for me for testing purposes, since I wasn't shure, if something should be happening in the other case. Since in the else part nothing is happening and the if part has a log, we also could just remove the whole else part.

Reply:
Makes sense. If it should be unreachable, it's probably a good idea to put an abort in there, although it doesn't seem like it should be a failure case. I'll have to give it another look.